docs/discovering_thanos_components.md

```diff
@@ -0,0 +1,61 @@
+# Discovering Thanos Components
+
+Thanos is a set of components. In order to successfully run Thanos these components need to be able to talk to each other. 
+More specifically:
+* `Thanos Query` needs to know about `Thanos Store` API servers in order to query metrics from them.
```

Comment:
This is super confusing because Thanos Store indicates stores gateway. Let's name it strictly `StoreAPI` and put link to StoreAPI proto definition maybe? (: